Updating to Version 0.3.0

emarinier opened the pull request:
Updates the IRIDA AMR detection plugin from version 0.2.0 to 0.3.0.

- Updates staramr to version 0.9.1
- Updates shovill to version 1.1.0
- Updates RGI to version 5.2.1
- RGI 5.2.1 is difficult to install with Conda (because of dependency resolution) and may need a manual installation using mamba.

apetkau commented on src/main/resources/workflows/0.3.0/irida_workflow.xml:
```diff
@@ -0,0 +1,145 @@
+    <parameter name="rgi-3-include_loose" defaultValue="false">
+      <toolParameter toolId="toolshed.g2.bx.psu.edu/repos/card/rgi/rgi/5.2.1" parameterName="include_loose"/>
+    </parameter>
+    <parameter name="rgi-3-rgi_db_local" required="true">
```
If no database is installed yet I get this error:

![image](https://user-images.githubusercontent.com/200517/207947522-187898a2-1d31-4144-8b08-5a769e63e109.png)

This is more of an issue for IRIDA itself, where it should give a warning like *"No required entries found from Galaxy tool data table"* rather than just crashing.

emarinier replied:
Is this solved by making the change below (db_opts.rgi_db_local), or is there something more nuanced that would need to be done?

apetkau replied:
This can't be solved here. It must be solved in the IRIDA codebase. I just made a comment here but there's nothing for you to do.

apetkau commented on src/main/resources/workflows/0.3.0/irida_workflow.xml:
```diff
@@ -0,0 +1,145 @@
+        </dynamicSource>
+        <toolParameter
+                toolId="toolshed.g2.bx.psu.edu/repos/card/rgi/rgi/5.2.1"
+                parameterName="card" />
```
The value `"card"` should be replaced with `"db_opts.rgi_db_local"` (this is what worked for me).

emarinier replied:
Is the "card" database (the one created using the data manager in Galaxy) not specified here? Is this pulling the names dynamically? Should there be a default option set somehow?

apetkau replied:
This pulls the database names dynamically based on the name assigned when installing the database in Galaxy. For me, for example, I named it `card-2022-12-15`.

The `parameterName` here is the name of the parameter that sets the database to use (and so the value should be `db_opts.rgi_db_local`).

emarinier replied:
Gotcha. Updated in 349871a.

apetkau reviewed (changes requested):
Thanks so much @emarinier, this looks amazing 😄 . It works great. I was successfully able to run some example genomes through the updated pipeline. I know there were some challenges getting dependencies installed and sorting out configuration/XML entries, but it's awesome.

I did have a few in-line comments for you. Another more general comment is: could you update both the `README.md` and `CHANGELOG.md` files to include information about the new version (such as the necessary tool versions to install in Galaxy)? Updating the `README.md` could also be done as another PR if you wanted too (though it's nice to update the `CHANGELOG.md` with the changes in this same PR).

apetkau commented:
Another comment I could make, is that for some reason increasing the number of cores assigned to shovill for me caused spades to crash (ran out of memory). I suspect this is more of a shovill (or spades) issue rather than an issue with your additions.

Also, here's, in brief, what I had to do to get rgi installed:

1. Run the following in the Galaxy machine (using the Galaxy conda) prior to installing the rgi tool in Galaxy:

   ```bash
   conda install -c conda-forge -c bioconda -c defaults mamba -y
   mamba create -c iuc -c conda-forge -c bioconda -c defaults --no-channel-priority --name __rgi@5.2.1 rgi=5.2.1 -y
   ```
2. Install RGI in Galaxy as normal

Maybe this could be added to the `README.md` too?

emarinier commented:
> I did have a few in-line comments for you. Another more general comment is: could you update both the `README.md` and `CHANGELOG.md` files to include information about the new version (such as the necessary tool versions to install in Galaxy)? Updating the `README.md` could also be done as another PR if you wanted too (though it's nice to update the `CHANGELOG.md` with the changes in this same PR).

I think I've everything that needed to be updated in the README and CHANGELOG:

- CHANGELOG in 09a6d78
- README in 6b23cd5

apetkau commented on README.md:
```diff
@@ -27,17 +27,17 @@ This project contains an [IRIDA][] plugin for a pipeline to perform AMR detectio
 Please download the provided `amr-detection-[version].jar` from the [releases][] page and copy to your `/etc/irida/plugins` directory.  Now you may start IRIDA and you should see the pipeline appear in your list of pipelines.
 
-*Note:* This plugin requires you to be running IRIDA version >= `21.01`. Please see the [IRIDA][] documentation for more details.
+*Note:* This plugin requires you to be running IRIDA version >= `22.09.4`. Please see the [IRIDA][] documentation for more details.
```
This is actually not correct. You still only need `21.01` IRIDA or greater. The value `22.09.4` you had set in the `pom.xml` file is only used for compile-time (not run-time).

emarinier replied:
Woops. Reverted in b612dd8

emarinier commented:
> Also, here's, in brief, what I had to do to get rgi installed:
> 
>     1. Run the following in the Galaxy machine (using the Galaxy conda) prior to installing the rgi tool in Galaxy:
>        ```shell
>        conda install -c conda-forge -c bioconda -c defaults mamba -y
>        mamba create -c iuc -c conda-forge -c bioconda -c defaults --no-channel-priority --name __rgi@5.2.1 rgi=5.2.1 -y
>        ```
> 
>     2. Install RGI in Galaxy as normal
> 
> 
> Maybe this could be added to the `README.md` too?

Added instructions in [09a6d78](https://github.com/phac-nml/irida-plugin-amr-detection/pull/18/commits/09a6d78acb7a83bbee0dcd1e0d2c54e6deb36069) and fixed some typos in [49f37db](https://github.com/phac-nml/irida-plugin-amr-detection/pull/18/commits/49f37db7818f3e738fb5be670b346e9d098a9270).

apetkau commented:
Thanks so much for all your changes Eric. This looks awesome.

The only other general comment that I forgot about before, is I'm wondering if you could add information in the README file about installing the RGI database? Possibly include a screenshot of the database installation tool?

emarinier commented:
> Thanks so much for all your changes Eric. This looks awesome.
> 
> The only other general comment that I forgot about before, is I'm wondering if you could add information in the README file about installing the RGI database? Possibly include a screenshot of the database installation tool?

Updated!

apetkau commented:
Thanks @emarinier . Looks great. Merging this in.
